### DISCUSSION TITLE
How to disable changing pages on swipe?

### DISCUSSION THREAD

**Discussion by JoeRakhimov** [readium/kotlin-toolkit#541] [Q&A] [answered]
I am developing an application based on the test-app. I want to disable changing pages on swipe and allow page navigation only on click events. The reason for this requirement is to enable text selection without the need for a long click.

I attempted to comment out the following lines in the EpubNavigatorFragment, but it did not disable page swiping:
webView.scrollToStart()
webView.scrollToEnd()
webView.setCurrentItem(0, false)
webView.setCurrentItem(webView.numPages - 1, false)

Could you please advise on how to disable page changing via swipe?

**Comment by mickael-menu** [comment, marked as answer]
It's not possible out of the box, but I would try injecting some JavaScript to add a `overflow: hidden` property on `body` or `html`. You will probably need some anyway to support the text selection without a long click.

**Reply by JoeRakhimov** [reply]
Injected JavaScript code that removes the style 'overflow: visible !important' from the page header and adds 'overflow: hidden' to body:

EpubNavigatorFragment:
```
is RunScriptCommand.Scope.WebView -> {
    command.scope.webView.runJavaScript(command.script);
    command.scope.webView.runJavaScript("document.head.childNodes[5].remove();");
    command.scope.webView.runJavaScript("document.body.style.overflow = 'hidden';");
}
```

As a result, page-swiping functionality is now disabled.
